### PR TITLE
Add GitHub CI for building to DockerHub

### PR DESCRIPTION
**Pull request by blinkdog**
Adds some .yml files for:
- Building a Docker image on commits to a Pull Request.
- Pushing a Docker image to DockerHub on commits pushed to `master`

Adds semantic versioning release based on @ric-evans work on [rest-tools](https://github.com/WIPACrepo/rest-tools/pull/17)

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -30,3 +30,11 @@ directory = htmlcov
 
 [flake8]
 ignore=D403,E226,E302,E305,E501,W503,W504
+
+[semantic_release]
+version_variable = rest_tools/__init__.py:__version__
```
Comment (ric-evans) [inline setup.cfg:35]: you need `lta` here

**Review by ric-evans — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 1 above)
Comment (ric-evans) [reply setup.cfg:35]: also, the update to __init__.py's version tuple :smile:

Comment (blinkdog) [reply setup.cfg]: I pushed some changes for that. Does it look good now?

Comment (ric-evans) [reply setup.cfg]: looks good :+1:

**Review by ric-evans — APPROVED** [review]

**Pull request merged**